Lib/ssl.py

```diff
@@ -876,6 +876,20 @@ def getpeercert(self, binary_form=False):
         """
         return self._sslobj.getpeercert(binary_form)
 
+    def get_verified_chain(self):
+        """Returns verified certificate chain provided by the other
+        end of the SSL channel as a list of ``_ssl.Certificate``.
```

Comment:
Digging into things here, this is the first time the internal `_ssl.Certificate` type is exposed as a public API. It has been kept as an internal implementation detail in the past.

The existing API returning a certificate is `getpeercert()` which returns the `dict` form that can be gotten via the `_ssl.Certificate.get_info()` method.  Returning a list of such `dict`s might make more sense here so that our internal implementation detail stays internal and this API behavior is consistent with that of `getpeercert()` by using a `dict` per certificate.

I think that'd just require turning this Python code into something resembling

```python
return [cert.get_info() for cert in self._sslobj.get_verified_chain()]
```

I'm not necessarily a fan of the whole `dict` thing as a design, but avoiding expanding our `ssl` module public API surface to expose the internal `_ssl.Certificate` class API public seems like a good thing without a clear reason to do so.

_(The other return type `getpeercert()` offers is its `binary_mode=True` DER-encoded bytes format, but I doubt that is what people really want a list of for a chain so I'd leave additional format options as future feature requests, only return the dict form in this PR)._

Reply:
That makes sense, but let me check first if the dict returned here will be enough to recreate `x509.Certificate` from `cryptography`.

Reply:
Unfortunately, this is impossible and we have to go with the `binary_mode=True` or twin method that will return certificates in DER format because it does not contain enough information and will be extremely hard to use.

This way each certificate can be turned into a `cryptography.x509.Certificate` by `cryptography.x509.load_der_x509_certificate`, so we can use it to build OCSP requests and validate OCSP responses. Or at least from the binary format, there is a way to do everything by hand.

Reply:
I implemented only binary mode, if we need both for come reason let me know I can do that as well.